### PR TITLE
Show commit message box in red if first line > 50 characters

### PR DESCRIPTION
**Pull request by WizardOhio24**
Linked to #478

Shows commit box in red if the first line is greater than 50, this can be configured in the theme and can be turned off entirely but 50 is the current default (Should it be off by default?).  If there were more options, there should probably be a separate config file, but because this is aesthetic only (and this would be the only config option) it could fit in theme.  

GitUI currently does not support making multi-line commit messages in the commit message box, so I'm not sure if there is a point in making the box red for multi-line messages which are unconventional (such as no empty 2nd line or >72 characters in the next paragraph as suggested in #478).

Screenshot_20210208_201559
Screenshot_20210208_201644

### CODE REVIEW

**Comment by extrawurst** [comment]
1) as stated in #478 I am not a fan of this requiring new configuration, especially since we now abuse the style file for it.
2) making the entire box red is too much. the count in the corner makes more sense
3) the unofficial limit of 50 only applies to the first line. we show the total char count, my suggestion would be to show an additional text on the bottom saying `first line exceeds X chars`

so long story short my required MVP for this:
* we hardcode the first-line-limit to 50 
* we show a red message on the bottom (of the dialog, left of the count) if the limit is exceeded

for the time being I mark this PR as *draft*

**Comment by WizardOhio24** [comment]
Now shows like this:
Screenshot_20210429_233225
:

Let me know what you think (I think this is visible enough, but not too visible). (This PR is still a draft, just have to clean up some code, I'll let you know when it's ready for review).

**Comment by extrawurst** [comment]
@WizardOhio24 still the whole message length is counted:

image

> so long story short my required MVP for this:
>
> * we hardcode the **first-line-limit** to 50
> * we show a red message on the bottom (of the dialog, left of the count) if the limit is exceeded
